perun-core/src/main/java/cz/metacentrum/perun/core/impl/DatabaseManagerImpl.java

```diff
@@ -73,4 +81,105 @@ public String getDatabaseInformation(PerunSession sess) throws InternalErrorExce
 			throw new InternalErrorException(ex);
 		}
 	}
+
+	public String getCodeDatabaseVersion(List<DBVersion> dbVersions, String currentDBVersion) {
+		if(dbVersions.isEmpty()) return currentDBVersion;
+		return dbVersions.get(0).getVersion();
+	}
+
+	public void updateDatabaseVersion(List<DBVersion> dbVersions) throws InternalErrorException {
+		Collections.reverse(dbVersions);
+
+		for (DBVersion v : dbVersions) {
+			log.debug("Executing update commands of version " + v.getVersion());
+			List<String> successfulCommands = new ArrayList<>();
+			for (String c : v.getCommands()) {
+				try {
+					jdbc.execute(c);
+					log.debug("Command executed: " + c);
+					successfulCommands.add(c);
+				} catch(EmptyResultDataAccessException ex) {
+					log.error("Update unsuccessful. All versions before " + v.getVersion() + " were successfully executed. " +
+							"Error executing command in version " + v.getVersion() + ": " + c, ex);
+					log.error("Successful commands from " + v.getVersion() + ": " + successfulCommands);
+					throw new ConsistencyErrorException("Update unsuccessful. Error executing command in version " + v.getVersion() + ": " + c, ex);
+				} catch(RuntimeException ex) {
+					log.error("Update unsuccessful. All versions before " + v.getVersion() + " were successfully executed. " +
+							"Error executing command in version " + v.getVersion() + ": " + c, ex);
+					log.error("Successful commands from " + v.getVersion() + ": " + successfulCommands);
+					throw new InternalErrorException("Update unsuccessful. Error executing command in version " + v.getVersion() + ": " + c, ex);
+				}
+			}
+		}
+	}
+
+	public List<DBVersion> getChangelogVersions(String currentDBVersion, String fileName) throws InternalErrorException {
+
+		Pattern versionPattern = Pattern.compile("^[1-9][0-9]*[.][0-9]+[.][0-9]+");
+		Pattern commentPattern = Pattern.compile("^--.*");
+
+		List<DBVersion> versions = new ArrayList<>();
+		boolean versionFound = false;
+
+		try(BufferedReader br = new BufferedReader(new InputStreamReader(ClassLoader.getSystemResourceAsStream(fileName)))) {
```

Comment:
After testing on devel I found a problem with ClassLoader in WAR file. In the tests there is just jar file which has no problem with defined classloader.
In WAR file need to be specific which ClassLoader will be used.
Use this instead:
Thread.currentThread().getContextClassLoader().getResourceAsStream(fileName)